.github/workflows/release.yml

```diff
@@ -0,0 +1,186 @@
+          sed -i -e "s|${{ env.downloadLocation }}/releases/download/latest|${{ env.downloadLocation }}/releases/download/${{ env.XML_SERVER_VERSION }}|g" package.json
+      - name: Build vscode-xml
+        run: |
+          npm install
+          npm run vscode:prepublish
+      - run: |
+          mkdir server/
+          if [ -e org.eclipse.lemminx*-uber.jar ]; then
+            cp org.eclipse.lemminx*-uber.jar server/
+          else
+            cp ../staging/org.eclipse.lemminx*-uber.jar server/
+          fi
+      - name: Prepare Pre-Release
+        if: ${{ github.event_name == 'schedule' || inputs.publishPreRelease == 'true' }}
+        run: |
+          npx gulp prepare_pre_release
+          echo "EXT_VERSION=$(cat package.json | jq -r .version)" >> $GITHUB_ENV
+          echo "publishPreReleaseFlag=--pre-release" >> $GITHUB_ENV
+      - name: Download LemMinX Binary Artifacts
+        uses: actions/download-artifact@v3
+      - name: Prepare Binary Artifacts For Packaging
+        run:  |
+          mkdir -p ../staging/zips ../staging/checksums
+          for f in lemminx-linux lemminx-osx-x86_64 lemminx-win32; do
+            pushd ${f}
+            chmod u+x ${f}*
+            zip ../../staging/zips/${f}.zip ${f}*
+            sha256sum ${f}* > ../../staging/checksums/${f}.sha256
+            mv ${f}* ../../staging
+            popd
+          done
+          ls -lashR ../staging
+      - name: Package vscode-xml
+        run: |
+          declare -A targets
+          targets["linux-x64"]=linux
+          targets["win32-x64"]=win32
+          targets["darwin-x64"]=osx-x86_64
+          targets["darwin-arm64"]=osx-x86_64
+          for tg in ${!targets[@]}; do
+            cp ../staging/lemminx-${targets[${tg}]}* ./server
+            cp ../staging/checksums/lemminx-${targets[${tg}]}.sha256 ./server
+            vsce package ${{ env.publishPreReleaseFlag }} --target ${tg} -o vscode-xml-${tg}-${{ env.XML_SERVER_VERSION }}-${GITHUB_RUN_NUMBER}.vsix
+            rm ./server/lemminx-*
+          done
+          cp ../staging/checksums/lemminx-*.sha256 ./server
+          vsce package -o vscode-xml-${{ env.XML_SERVER_VERSION }}-${GITHUB_RUN_NUMBER}.vsix
+          ls -lash *.vsix
+      - name: Upload VSIX Artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: vscode-xml
+          path: vscode-xml-*-${{ env.XML_SERVER_VERSION }}-${{github.run_number}}.vsix
+          if-no-files-found: error
+      - name: Upload LemMinX Uber Jar
+        uses: actions/upload-artifact@v2
+        with:
+          name: lemminx-uber-jar
+          path: server/org.eclipse.lemminx*-uber.jar
+          if-no-files-found: error
+      - name: Publish to GH Release Tab
+        if: ${{ inputs.publishToMarketPlace == 'true' && inputs.publishToOVSX == 'true' }}
+        uses: "marvinpinto/action-automatic-releases@latest"
```

Comment:
This GH Action version should be pinned to the sha-1 commit as it's not a trusted publisher.